### PR TITLE
Fix tests and static analysis

### PR DESCRIPTION
**Pull request by derrabus**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no
| Fixed issues | N/A

#### Summary

This PR attempts to fix all current CI failures.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
     "require": {
         "php": "^7.4 || ^8.0",
         "composer-runtime-api": "^2",
-        "doctrine/dbal": "^3.3",
+        "doctrine/dbal": "^3.4",
```
Comment (derrabus) [inline composer.json]: I hope this bump to DBAL 3.4.3 is okay for a bugfix release. If not, I could revert here and skip the three failing tests on the low-deps run.

File: lib/Doctrine/Migrations/Configuration/Migration/XmlFile.php [hunk 2]
```diff
@@ -87,10 +87,7 @@ private function extractParameters(SimpleXMLElement $root, bool $loopOverNodes):
             } elseif ($nodeName === 'storage' && $node->{'table-storage'} instanceof SimpleXMLElement) {
                 $config['table_storage'] = $this->extractParameters($node->{'table-storage'}, false);
             } elseif ($nodeName === 'migrations') {
-                $config['migrations'] = [];
-                foreach ($node->{'migration'} as $pathNode) {
-                    $config['migrations'][] = (string) $pathNode;
-                }
+                $config['migrations'] = $this->extractMigrations($node);
```
Comment (derrabus) [inline lib/Doctrine/Migrations/Configuration/Migration/XmlFile.php:90]: Workaround for phpstan/phpstan#8015

File: tests/Doctrine/Migrations/Tests/Tools/Console/Command/MigrateCommandTest.php [hunk 3]
```diff
@@ -422,7 +422,7 @@ public function testExecuteMigrateCancelExecutedUnavailableMigrations(): void
         $output = $this->migrateCommandTester->getDisplay(true);
 
         self::assertStringContainsString('[WARNING] You have 1 previously executed migrations in the database that are not registered migrations.', $output);
-        self::assertStringContainsString('WARNING! You are about to execute a migration in database "<unnamed>" that could result in schema changes and data loss. Are you sure you wish to continue?', $output);
+        self::assertStringContainsString('WARNING! You are about to execute a migration in database "main" that could result in schema changes and data loss. Are you sure you wish to continue?', $output);
```
Comment (derrabus) [inline tests/Doctrine/Migrations/Tests/Tools/Console/Command/MigrateCommandTest.php:425]: On DBAL 3.4, the default schema on SQLite is `main` which is more correct than the empty string that was reported previously.

**Review by derrabus — COMMENTED** [review]
💡 Let me explain my changes quickly.

File: composer.json (hunk 1 above)
Comment (stof) [reply composer.json]: if the supported PHP versions have not changed between DBAL 3.3 and 3.4 (or if the dropped PHP versions were already not supported here), it is fine IMO.

**Review by goetas — APPROVED** [review]

**Review by stof — APPROVED** [review]

File: composer.json (hunk 1 above)
Comment (stof) [reply composer.json]: And the answer to that is that DBAL 3.4 bumped the min PHP version from 7.3 to 7.4, which is fine as doctrine/migrations already requires PHP 7.4 anyway.

File: phpstan.neon.dist [hunk 4]
```diff
@@ -24,6 +24,11 @@ parameters:
             message: '~^Method Doctrine\\Migrations\\Tests\\Stub\\DoctrineRegistry::getService\(\) should return Doctrine\\Persistence\\ObjectManager but returns Doctrine\\DBAL\\Connection\|Doctrine\\ORM\\EntityManager~'
             path: tests/Doctrine/Migrations/Tests/Stub/DoctrineRegistry.php
 
+        # We cannot document the closure as precise as PHPStan wants us to.
```
Comment (greg0ire) [inline phpstan.neon.dist:27]: Why not? I'm on my phone and cannot easily check whether it is possible to document a closure, but it seems possible with`@var`? Or maybe with an inline comment?

Comment (derrabus) [reply phpstan.neon.dist:27]: If you find a nice way to do so, be my guest. The closure that we're passing here looks fine from my POV, so I'm not sure if this error is really worth fixing.

**Pull request merged**